sdk/typescript/src/providers/json-rpc-provider.ts

```diff
@@ -54,7 +99,110 @@ export class JsonRpcProvider extends Provider {
     public skipDataValidation: boolean = false
   ) {
     super();
+
     this.client = new JsonRpcClient(endpoint);
+
+    this.wsEndpoint = getWebsocketUrl(endpoint);
+
+    const socketOptions = { reconnect_interval: 3000, autoconnect: false };
+    this.wsClient = new WsRpcClient(this.wsEndpoint, socketOptions);
+    this.setupSocket();
+  }
+
+  private setupSocket() {
+    this.wsClient.on('open', () => {
+      if(this.wsTimeout != 0) {
+        clearTimeout(this.wsTimeout);
+        this.wsTimeout = 0;
+      }
+      this.wsConnectionState = ConnectionState.Connected;
+      // underlying websocket is private, but we need it
```

Comment:
I'd open an issue about this on the `rpc-websockets` repo. This could easily be refactored in a patch update and break our client.